src/Core/src/Handlers/View/ViewHandler.cs

```diff
@@ -198,7 +197,20 @@ public static void MapVisibility(IViewHandler handler, IView view)
 
 		public static void MapBackground(IViewHandler handler, IView view)
 		{
-			((PlatformView?)handler.PlatformView)?.UpdateBackground(view);
+			if (handler.PlatformView is not PlatformView platformView)
+				return;
+
+			if (view.Background is ImageSourcePaint image)
+			{
+				var provider = handler.GetRequiredService<IImageSourceServiceProvider>();
+
+				platformView.UpdateBackgroundImageSourceAsync(image.ImageSource, provider)
+					.FireAndForget(handler);
+			}
```

Comment:
This if block here is not so great, but still is good enough. Since this logic lives here and not in the extension method, none of the controls that re-define the background mapping - such as image, label, entry - get this enhanced background support.

To fully realize this new feature, we will need to move this logic into the `UpdateBackground` method and make it async and all that. Also, we probably need to consolidate the several `UpdateBackground` methods and variants scattered about.